### DISCUSSION TITLE
Support multiple versions of PrimeNG in SPA(Angular)

### DISCUSSION THREAD

**Discussion by huzurpolat** [primefaces/community#199] [PrimeNG] [answered]
Hi Guys,

we are looking for a solution to use Multiple versions of PrimeNG in Angular context.

Is it generally possible to have multiple versions of Prime Ng in SPA app?
i found this thread in PrimeNg Forum: https://forum.primefaces.org/viewtopic.php?t=51509 
Is it still relevant?

THX

**Comment by cetincakiroglu** [comment, marked as answer]
To run multiple versions, for example, 3 versions of PrimeNG in an app, you need 3 different apps to shrink into 1 app, and 3 different builds and find a way to manage them. It's not a good practice to build 3 different versions and we don't suggest it. And in order to use different versions of PrimeNG you need to use relevant Angular versions as well because PrimeNG versions are synced with Angular versions, for example 

Angular 15 -> PrimeNG 15

Angular 16 -> PrimeNG 16

Angular does not fully support all the previous versions. There may be some deprecations and breaking changes between versions resulting in malfunction of components.
